lib/grape/validations/params_scope.rb

```diff
@@ -140,9 +140,14 @@ def require_optional_fields(context, opts)
       end
 
       def validate_attributes(attrs, opts, &block)
```

Comment:
Have you tried to do this?

```
def validate_attributes(attrs, validations, opts = {}, &block)
```

This way there wouldn't be any reason to do the selection below.